cluster/addons/dns/kube-dns/transforms2salt.sed

```diff
@@ -1,5 +1,5 @@
-s/__PILLAR__DNS__SERVER__/{{ pillar['dns_server'] }}/g
-s/__PILLAR__DNS__DOMAIN__/{{ pillar['dns_domain'] }}/g
-s/__PILLAR__CLUSTER_CIDR__/{{ pillar['service_cluster_ip_range'] }}/g
-s/__PILLAR__DNS__MEMORY__LIMIT__/{{ pillar['dns_memory_limit'] }}/g
+s/__DNS__SERVER__/{{ pillar['dns_server'] }}/g
+s/__DNS__DOMAIN__/{{ pillar['dns_domain'] }}/g
+s/__CLUSTER_CIDR__/{{ pillar['service_cluster_ip_range'] }}/g
+s/__DNS__MEMORY__LIMIT__/{{ pillar['dns_memory_limit'] }}/g
```

Comment:
this left pillar references in the file, but stopped replacing them in setup-kube-dns-manifest

Reply:
I think that this issue was caught in the CI and fixed by https://github.com/kubernetes/kubernetes/pull/95580

Reply:
nope, setup-kube-dns-manifest is still wrong in master:

https://github.com/kubernetes/kubernetes/blob/c3c050c88f4265ba77464682801e2cc1ab8021f7/cluster/gce/gci/configure-helper.sh#L2492-L2495

Reply:
@liggitt the configure-helper.sh only looks at kube-dns.yaml.in which has no "pillar" references anymore.
https://github.com/kubernetes/kubernetes/blob/c3c050c88f4265ba77464682801e2cc1ab8021f7/cluster/addons/dns/kube-dns/kube-dns.yaml.in#L33

so the reference string replace above should be enough.
https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/kube-dns/transforms2salt.sed still has some "pillar" references, but I don't think that is used anywhere.

 The issue @aojea is referring to was specific to nodelocaldns, which got fixed. 
Here is CI that is deploying kube-dns on a GCE k8s cluster.